src/components/Picker/BasePicker.tsx

```diff
@@ -0,0 +1,194 @@
+    const root = useRef<View>(null);
+
+    // reference to @react-native-picker/picker
+    const picker = useRef<RNPickerSelect>(null);
+
+    // Windows will reuse the text color of the select for each one of the options
+    // so we might need to color accordingly so it doesn't blend with the background.
+    const pickerPlaceholder = Object.keys(placeholder).length ? {...placeholder, color: themeColors.pickerOptionsTextColor} : {};
+
+    useEffect(() => {
+        if (!!value || !items || items.length !== 1 || !onInputChange) {
+            return;
+        }
+
+        // When there is only 1 element in the selector, we do the user a favor and automatically select it for them
+        // so they don't have to spend extra time selecting the only possible value.
+        onInputChange(items[0].value, 0);
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [items]);
+
+    const context = useScrollContext();
+
+    /**
+     * Forms use inputID to set values. But BasePicker passes an index as the second parameter to onValueChange
+     * We are overriding this behavior to make BasePicker work with Form
+     */
+    const onValueChange = (inputValue: TPickerValue, index: number) => {
+        if (inputID) {
+            onInputChange(inputValue);
+            return;
+        }
+
+        onInputChange(inputValue, index);
+    };
+
+    const enableHighlight = () => {
+        setIsHighlighted(true);
+    };
+
+    const disableHighlight = () => {
+        setIsHighlighted(false);
+    };
+
+    useImperativeHandle(ref, () => ({
+        /**
+         * Focuses the picker (if configured to do so)
+         *
+         * This method is used by Form
+         */
+        focus() {
+            if (!shouldFocusPicker) {
+                return;
+            }
+
+            // Defer the focusing to work around a bug on Mobile Safari, where focusing the `select` element in the
+            // same task when we scrolled to it left that element in a glitched state, where the dropdown list can't
+            // be opened until the element gets re-focused
+            lodashDefer(() => {
+                picker.current?.focus();
+            });
+        },
+
+        /**
+         * Like measure(), but measures the view relative to an ancestor
+         *
+         * This method is used by Form when scrolling to the input
+         *
+         * @param relativeToNativeComponentRef - reference to an ancestor
+         * @param onSuccess - callback called on success
+         * @param onFail - callback called on failure
+         */
+        measureLayout(relativeToNativeComponentRef, onSuccess, onFail) {
+            if (!root.current) {
+                return;
+            }
+
+            root.current.measureLayout(relativeToNativeComponentRef, onSuccess, onFail);
+        },
+    }));
+
+    const hasError = !!errorText;
+
+    if (isDisabled) {
+        return (
+            <View key={inputID}>
```

Comment:
Why did you add `key={inputID}`? afaik `inputID` is only used internally inside `Form` logic.

Reply:
Initially there was a `forwardRef` which also added an[ inputID key](https://github.com/Expensify/App/pull/30646/files#diff-e593b5840fcd595845413e15b71f9db2645946ed22118c13c22f5c0fbd096043L286) to the element.

But during [this discussion](https://github.com/Expensify/App/pull/30646#discussion_r1385181815) it was updated.

@fabioh8010 So do you think it makes sense to revert it all (including `forwardRef` logic)?

Reply:
Yeah I think it would be safer to revert it 🤔